### ISSUE TITLE
Confusing error message when `cd` to non-existing directory

### ISSUE THREAD

**Issue by Hofer-Julian** [nushell/nushell#8027]
### Describe the bug

If I want to `cd` into a non-existing directory, the error message adds confusing implementation details.

### How to reproduce

```
nu ❯ cd askldjflasjk
Error: nu::shell::directory_not_found (link)

  × Directory not found
   ╭─[entry #1:1:1]
 1 │ cd askldjflasjk
   ·    ──────┬─────
   ·          ╰── directory not found
   ╰────
  help: IO Error: NotADirectory(Span { start: 85214, end: 85226 })
```

### Expected behavior

The same thing but without `NotADirectory(Span { start: 85214, end: 85226 })`

### Configuration

| key                | value                                    |
| ------------------ | ---------------------------------------- |
| version            | 0.75.0                                   |
| branch             |                                          |
| commit_hash        | 17a265b197ad6bb77db5a07942a9f160ec260eb0 |
| build_os           | windows-x86_64                           |
| build_target       | x86_64-pc-windows-msvc                   |
| rust_version       | rustc 1.66.1 (90743e729 2023-01-10)      |
| rust_channel       | 1.66.1-x86_64-pc-windows-msvc            |
| cargo_version      | cargo 1.66.1 (ad779e08b 2023-01-10)      |
| pkg_version        | 0.75.0                                   |
| build_time         | 2023-01-31 20:39:29 +00:00               |
| build_rust_channel | release                                  |
| features           | database, default, trash, which, zip     |
| installed_plugins  |                                          |

**Comment by fdncred** [comment]
I think it's fine to have the IO error there like:
```
help: IO Error: NotADirectory
```
but not the span. Good catch.

**Comment by ryand67** [comment]
I'd like to take a stab at this one if it's available!

**Comment by fdncred** [comment]
Go for it @ryand67. Thanks for jumping in.

**Comment by rgwood** [comment]
Fixed in #8049
